idr/attributes.py

```diff
@@ -170,16 +174,217 @@ def get_phenotypes_for_gene(gene_name, screenid=None):
+
+            accnames = list(accname.values)
+            accids = list(accid.values)
+            idlist = []
+            for id in scrid:
+                idx = id.index('_')
+                idlist.append(id[:idx])
+
+            for idx, idx1 in enumerate(accnames):
+                phenotypenametoacc[accnames[idx]] = accids[idx]
+
+            totalphenotypename = totalphenotypename + accnames
+            totalphenotypeaccession = totalphenotypeaccession + accids
+            totalscreenids = totalscreenids + list(scrid.values)
+
+            genedict[gene] = [entrezid, ensembleid,
+                              None, None, None, None, None]
+            genedict[gene][2] = key
+            genedict[gene][3] = gid
+            genedict[gene][4] = accnames
+            genedict[gene][5] = accids
+            genedict[gene][6] = idlist
+
+        progress(ids+1, len(go_gene_list),
+                 status='Iterating through gene list')
+
+    query_genes_dataframe = pandas.DataFrame.from_dict(genedict,
+                                                       orient='index')
+    query_genes_dataframe.columns = ("Entrez", "Ensembl",
+                                     "Key", "Value", "PhenotypeName",
+                                     "PhenotypeAccession", "ScreenIds")
+
+    # get the screens to phenotypes map for the query genes
+    organism_screen_idlist = get_organism_screenids(idr_base_url,
+                                                    session, organism)
+    genes_scid_list = [item for sublist in
+                       query_genes_dataframe['ScreenIds'].values
+                       for item in sublist]
+    genes_scid_list = list(set(genes_scid_list))
+    screen_to_phenotype_dictionary = {}
+    for scid in genes_scid_list:
+        if scid in organism_screen_idlist:
+            content = [x for x in set(list(totalscreenids))
+                       if x.startswith(scid)]
+            for idx, item in enumerate(content):
+                idx1 = item.index('_')
+                content[idx] = item[idx1+1:]
+            screen_to_phenotype_dictionary[scid] = content
+
+    return [query_genes_dataframe, screen_to_phenotype_dictionary]
+
+
+def get_similar_genes(conn, query_genes_list, screen_to_phenotype_dictionary):
+
+    """
+    Return a multi-dimensional dictionary with the
+    following mapping,
+    similar_genes[screenid][phenotypename] = similar_genes_list
+    (uses python blitz gateway)
+    """
+
+    similar_genes = {}
+    overlap_genes = {}
+    scid_list = set(list(screen_to_phenotype_dictionary.keys()))
+    for i, sid in enumerate(set(scid_list)):
+
+        phlist = screen_to_phenotype_dictionary[sid]
+        similar_genes[str(sid)] = {}
+        overlap_genes[str(sid)] = {}
+        for phenotype in np.unique(phlist):
+
+            args = {
+                "name": "Phenotype Term Name",
+                "value": phenotype,
+                "ns": "openmicroscopy.org/mapr/phenotype",
```

Comment:
`openmicroscopy.org/mapr/*` namespaces are useful to other users. Can you make them a top level variable instead, e.g. `idr.mapr.PHENOTYPE` or `idr.MAPR_PHENOTYPE` and similarly for all other mapr namespaces?

Reply:
Thinking about it, those namespaces should really have matched `/idr/`.